core/camel-support/src/main/java/org/apache/camel/support/PropertyBindingSupport.java

```diff
@@ -516,6 +516,32 @@ private static boolean doBindProperties(CamelContext camelContext, Object target
             uOptionPrefix = optionPrefix.toUpperCase(Locale.US);
         }
 
+        // we need to set references first
```

Comment:
Yeah that seems like a good idea as they are likely more complex objects than simple types for boolean, ints, strings etc. Then having those set first then chances are the others can be set, also if they are sub options on these objects that was set as reference parameters. Also when the order of properties are not correct, maybe the end user didnt set it right, so this is an improvement overall anyway.